### PR TITLE
Add a promo banner celebrating bitcoin 2022 conf

### LINKED ISSUE #1695: Taker UI "celebrating Bitcoin 2022 conference" banner
Banner that announces promo for Bitcoin2022

`Celebrating Bitcoin2022 in Miami`

Would be cool if we could do this with a request so we can announce different promos.

### PR DESCRIPTION
**Pull request by klochowicz**
Banner will be displayed by the takers between 04/04/2022-18/04/2022 (inclusive).

Links to testing.itchysats.network, which can be linked to a blog post when it
gets published.

screenshot:
image

### CODE REVIEW

**Comment by klochowicz** [comment]
I opted for the simplest approach in the end, without dynamically swapping the content of the promo - I wanted to make sure that we have this on time for the conference running on umbrel, we can have a generalised approach later on if there will be another promo coming up.

File: taker-frontend/src/App.tsx [hunk 1]
```diff
@@ -110,6 +112,11 @@ export const App = () => {
 
     dayjs.extend(relativeTime);
     dayjs.extend(utc);
+    dayjs.extend(isBetween);
+
+    // Show promo banner from 04.04.2022 until 18.04.2022 (specify midnight the
+    // next day as the end)
+    const isWithinPromoPeriod = dayjs().utc().isBetween("2022-04-04", "2022-04-19");
```
Comment (bonomat) [inline taker-frontend/src/App.tsx]: I propose to show this sooner. 
```suggestion
    const isWithinPromoPeriod = dayjs().utc().isBetween("2022-03-29", "2022-04-19");
```

**Comment by bonomat** [comment]
Do you have a screenshot on how it looks please?

**Comment by klochowicz** [comment]
> Do you have a screenshot on how it looks please?

added screenshot. I opted for simple, not-too-annoying text message.

**Comment by bonomat** [comment]
> > Do you have a screenshot on how it looks please?
> 
> added screenshot. I opted for simple, not-too-annoying text message.

Thanks. I'd be in favor of making this more prominent, e.g. by using an alertbox: 
```
<Alert status='info'>
    <AlertIcon />
    Celebrating ....
  </Alert>
```

see here for more choices: https://chakra-ui.com/docs/components/feedback/alert

**Comment by klochowicz** [comment]
> > > Do you have a screenshot on how it looks please?
> > 
> > 
> > added screenshot. I opted for simple, not-too-annoying text message.
> 
> Thanks. I'd be in favor of making this more prominent, e.g. by using an alertbox:
> 
> ```
> <Alert status='info'>
>     <AlertIcon />
>     Celebrating ....
>   </Alert>
> ```
> 
> see here for more choices: https://chakra-ui.com/docs/components/feedback/alert

great idea, thanks! this is how it looks wrapped inside an alert: 

image


I didn't add the AlertIcon, as it is used to convey severity of an alert, which is not applicable here IMHO.

**Comment by bonomat** [comment]
I like this more :) maybe add   behind the text so that one can see that it's a link :) 

I mean this icon: 
https://fontawesome.com/icons/arrow-up-right-from-square?s=solid

image

**Review by bonomat — APPROVED** [review]
👍

File: taker-frontend/src/App.tsx [hunk 2]
```diff
@@ -265,6 +279,18 @@ function HistoryLayout({ cfds, connectedToMaker }: HistoryLayoutProps) {
     </VStack>);
 }
 
+function PromoBanner() {
+    return (<HStack>
+        <Center>
+            <Alert status="info">
+                <Link href="http://testing.itchysats.network/" isExternal>
+                    🎁Celebrating pitching on Bitcoin2022 in Miami... 🎉
```
Comment (bonomat) [inline taker-frontend/src/App.tsx]: Is this correct? 

```suggestion
                    🎁Celebrating pitching at Bitcoin2022 in Miami... 🎉
```

Comment (klochowicz) [reply taker-frontend/src/App.tsx]: thank you, it was indeed a grammar error!

**Comment by klochowicz** [comment]
> I like this more :) maybe add  behind the text so that one can see that it's a link :)
> 
> I mean this icon: https://fontawesome.com/icons/arrow-up-right-from-square?s=solid
> 
> image

I'll submit a follow-up after this gets in that extracts a reusable Link component, as it's not the first time I forgot about this.

**Comment by klochowicz** [comment]
Force-pushed with a bit of refactor as I didn't want to import the icon at the top level app - now it's a separate component, but it looks and behaves as discussed above.

**Comment by klochowicz** [comment]
last screenshot:
image


note that that external icon is not horizontally aligned with the text link, but this is a problem in chakra. I will try to tackle that one when creating Link component.

File: taker-frontend/src/components/PromoBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import { Alert, Center, HStack, Link } from "@chakra-ui/react";
+
+import React from "react";
+
+export default function PromoBanner() {
+    return (<HStack>
+        <Center>
+            <Alert status="info">
+                <Link href="http://testing.itchysats.network/" isExternal>
+                    🎁Celebrating pitching at Bitcoin2022 in Miami... 🎉
```
Comment (bonomat) [inline taker-frontend/src/components/PromoBanner.tsx]: Might look better with a space 😬
```suggestion
                    🎁 Celebrating pitching at Bitcoin2022 in Miami... 🎉
```

**Review by bonomat — APPROVED** [review]

File: taker-frontend/src/components/PromoBanner.tsx (hunk 3 above)
Comment (klochowicz) [reply taker-frontend/src/components/PromoBanner.tsx]: agreed!

**Comment by klochowicz** [comment]
Bors r+

**Pull request merged**